homeassistant/components/sems_portal/sensor.py

```diff
@@ -0,0 +1,278 @@
+        self, coordinator: SemsDataUpdateCoordinator, name: str
+    ) -> Any:
+        """Retrieve the inverter by name."""
+
+        for inverter in coordinator.data["powerPlant"]["inverters"]:
+            if inverter["name"] == name:
+                return inverter
+        return None
+
+    def __init__(
+        self,
+        name: str,
+        model: str,
+        config_entry: ConfigEntry,
+        description: SensorEntityDescription,
+        coordinator: SemsDataUpdateCoordinator,
+    ) -> None:
+        """Initialize the sensor."""
+
+        self.coordinator = coordinator
+        self.deviceName = name
+        deviceName = self.deviceName
+        deviceModel = model
+        self._config_entry_id = config_entry.entry_id
+        self.entity_description = description
+        self._attr_unique_id = f"{deviceName}-{description.key}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, deviceName)},
+            manufacturer="Goodwe",
+            model=deviceModel,
+            name=deviceName,
+        )
+
+    @property
+    def native_value(self):
+        """Return the state of the sensor."""
+        return self.getInverterByName(self.coordinator, self.deviceName)[
+            self.entity_description.key
+        ]
+
+    async def async_added_to_hass(self) -> None:
+        """Connect to dispatcher listening for entity data notifications."""
+        self.async_on_remove(
+            self.coordinator.async_add_listener(self.async_write_ha_state)
+        )
+
+    async def async_update(self) -> None:
+        """Get the latest data from OWM and updates the states."""
+        await self.coordinator.async_request_refresh()
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Get the setup sensor."""
+
+    coordinator: SemsDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    interters = coordinator.data["powerPlant"]["inverters"]
+
+    inverterEntities = [
+        SemsInverterSensor(
+            inverter["name"], inverter["model"], config_entry, description, coordinator
+        )
+        for description in SENSOR_TYPES_INVERTERS
+        for inverter in interters
+    ]
+
+    powerPlantInformationEntities = [
+        SemsInformationSensor(config_entry, description, coordinator)
+        for description in SENSOR_TYPES_POWERSTATION
+    ]
+
+    async_add_entities(inverterEntities)
+    async_add_entities(powerPlantInformationEntities)
+
+
+SENSOR_TYPES_INVERTERS: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        name="Inner Temp",
+        key="innerTemp",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+    ),
+)
+
+SENSOR_TYPES_POWERSTATION: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        name="Powerstation Id",
```

Comment:
is it possible to use `translation_key` here and move the nae to `strings.json`? Would allow for better localization support (applies to all sensors)